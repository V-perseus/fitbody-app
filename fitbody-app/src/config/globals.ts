import { Dimensions, TextProps } from 'react-native'
import styles from './styles'
import string from './strings'
import { FB_CURRENT_ENV, FB_ENVIRONMENTS } from '../../env'

export const globals = {
  // Styles & Colors
  styles: {
    ...styles,
  },
  // Fonts
  fonts: {
    ...styles.fonts,
  },
  // Header Styles
  header: {
    icons: {
      marginRight: 15,
      marginLeft: 15,
    },
    headerTitleStyle: <TextProps>{
      fontSize: 16,
      fontFamily: styles.fonts.primary.semiboldStyle.fontFamily,
      textAlign: 'center',
    },
    headerTruncatedBackTitle: true,
  },
  // Standard String
  strings: {
    ...string,
  },
  // Background Images
  bgImage: {
    imageStyle: { resizeMode: 'cover' },
    style: {
      width: Dimensions.get('window').width,
      height: Dimensions.get('window').height,
    },
  },
  // Window dimensions
  window: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  },
  // Button click targeting
  touchableOpacity: {
    activeOpacity: 0.8,
    hitSlop: { top: 5, right: 5, left: 5, bottom: 5 },
  },
  fbEnvironments: {
    ...FB_ENVIRONMENTS,
  },
  currentEnv: FB_CURRENT_ENV,
  get apiBase(): string {
    return this.fbEnvironments[this.currentEnv]
  },
}

export default globals
