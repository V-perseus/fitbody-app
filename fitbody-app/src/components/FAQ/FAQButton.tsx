import React from 'react'
import { StyleSheet, TouchableOpacity, Text } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import globals from '../../config/globals'

interface IFAQButtonProps {
  active: boolean
  title: string
  handlePress: () => void
}
const FaqTab = ({ active, title, handlePress }: IFAQButtonProps) => (
  <TouchableOpacity onPress={handlePress} style={styles.submitContainer} testID="tab_btn">
    <LinearGradient
      style={active ? styles.tabActive : styles.tabInactive}
      testID="tab_bg"
      colors={
        active
          ? [globals.styles.colors.colorTopaz, globals.styles.colors.colorSkyBlue]
          : [globals.styles.colors.colorGray, globals.styles.colors.colorGray]
      }>
      <Text style={styles.buttonText}>{title}</Text>
    </LinearGradient>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  buttonText: {
    fontSize: 16,
    color: globals.styles.colors.colorWhite,
    fontFamily: globals.fonts.primary.boldStyle.fontFamily,
  },
  submitContainer: {
    paddingTop: 100,
  },
  tabActive: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 57,
    borderRadius: 50,
    width: 160,
    marginTop: 5,
    marginLeft: 9,
    marginRight: 9,
    marginBottom: 5,
  },
  tabInactive: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 45,
    borderRadius: 50,
    width: 140,
    marginTop: 10,
    marginLeft: 9,
    marginRight: 9,
    marginBottom: 5,
  },
})

export default FaqTab
