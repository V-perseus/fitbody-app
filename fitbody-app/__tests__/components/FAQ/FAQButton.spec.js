import React from 'react'
import { cleanup, fireEvent, render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import FAQButton from '../../../src/components/FAQ/FAQButton'
import globals from '../../../src/config/globals'

describe('FAQButton', () => {
  let handlePress
  beforeEach(() => {
    handlePress = jest.fn()
  })

  afterEach(cleanup)

  it('renders snapshot correctly', () => {
    const tree = renderer.create(<FAQButton active={true} title="TECH SUPPORT" handlePress={() => handlePress('support')} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render button', () => {
    const { queryByText } = render(<FAQButton active={true} title="TECH SUPPORT" handlePress={() => handlePress('support')} />)
    expect(queryByText('TECH SUPPORT')).not.toBeNull()
  })

  it('should render inactive state', () => {
    // active
    //   ? [globals.styles.colors.colorTopaz, globals.styles.colors.colorSkyBlue]
    //   : [globals.styles.colors.colorGray, globals.styles.colors.colorGray]
    const container = render(<FAQButton active={false} title="TECH SUPPORT" handlePress={() => handlePress('support')} />)
    expect(container.queryByTestId('tab_bg')).toBeDefined()
    expect(container.queryByText(globals.styles.colors.colorGray)).toBeDefined()
  })

  it('should render active state', () => {
    // active
    //   ? [globals.styles.colors.colorTopaz, globals.styles.colors.colorSkyBlue]
    //   : [globals.styles.colors.colorGray, globals.styles.colors.colorGray]
    const container = render(<FAQButton active={true} title="TECH SUPPORT" handlePress={() => handlePress('support')} />)
    expect(container.queryByTestId('tab_bg')).toBeDefined()
    expect(container.queryByText(globals.styles.colors.colorTopaz)).toBeDefined()
    expect(container.queryByText(globals.styles.colors.colorSkyBlue)).toBeDefined()
  })

  it('should handle press event', async () => {
    const { queryByTestId } = render(<FAQButton active={true} title="TECH SUPPORT" handlePress={() => handlePress('support')} />)
    expect(queryByTestId('tab_btn')).toBeDefined()
    await fireEvent.press(queryByTestId('tab_btn'))
    expect(handlePress).toHaveBeenCalledWith('support')
  })
})
