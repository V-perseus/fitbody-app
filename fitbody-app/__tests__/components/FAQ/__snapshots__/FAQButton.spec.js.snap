// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQButton renders snapshot correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "paddingTop": 100,
    }
  }
  testID="tab_btn"
>
  <BVLinearGradient
    colors={
      [
        4279220154,
        4286300671,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      {
        "alignItems": "center",
        "borderRadius": 50,
        "height": 57,
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": 9,
        "marginRight": 9,
        "marginTop": 5,
        "width": 160,
      }
    }
    testID="tab_bg"
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 16,
        }
      }
    >
      TECH SUPPORT
    </Text>
  </BVLinearGradient>
</View>
`;
